src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java

```diff
@@ -881,4 +902,30 @@ void setChannelFactory(ChannelFactory channelFactory) {
   RemoteActionContextProvider getActionContextProvider() {
     return actionContextProvider;
   }
+
+  private Credentials newCredentialsFromNetrc(CommandEnvironment env) throws IOException {
```

Comment:
Ideally, this method would be static and have tests. Consider changing the interface to take a client env map and FileSystem, which might be easier to test.

Reply:
Changed and added few tests.